### PR TITLE
fix: Mobile schedule menu

### LINKED ISSUE #5919: Mobile schedule: Fix menu
**Current State:**
Screenshot_20201204_230526_com android chrome

Please change it as follows:
**1. Version with not expanded search**
   1. Move the menu as follows
   2. Reduce the space to the schedule (also ensure there is not too much space if there are tracks)

101238106-2b2de600-36de-11eb-8637-7c46f38eadb4

**2. When the user clicks the search button, expand the search bar as in the below screenshot.**

101238106-2b2de600-36de-11eb-8637-7c46f38eadb4-a

### PR DESCRIPTION
**Pull request by ranjitkshah**
fixes #5919 

#### Short description of what this resolves:
enhancing the style of schedule sessions menu
image

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/tables/utilities/search-box.js [hunk 1]
```diff
@@ -17,4 +17,9 @@ export default class SearchBox extends Component {
     debounce(this, this.setSearchQueryLazily, value, this.debouncePeriod);
   }
 
+  @action
+  showSessionFields() {
+    document.querySelector('#session-search').classList.toggle('show-div');
```
Comment (iamareebjamal) [inline app/components/tables/utilities/search-box.js:22]: This is ember, not jquery

File: app/templates/components/tables/utilities/search-box.hbs [hunk 2]
```diff
@@ -1,4 +1,4 @@
 <div class="ui {{if (eq @size "large") '' 'small'}} icon input search-box">
-  <Input @type="text" @value={{readonly this.searchQuery}} @keyUp={{action 'setSearchQuery' value="target.value"}} placeholder="{{t "Search"}} ..." />
+  <Input  class="{{ if this.device.isMobile "expand-bar" "" }}" @type="text" @value={{readonly this.searchQuery}} @keyUp={{action 'setSearchQuery' value="target.value"}} onfocus={{ if this.device.isMobile (action "showSessionFields")}} @focus-out={{if this.device.isMobile (action "showSessionFields")}}  placeholder="{{t "Search"}} ..." />
```
Comment (sachinchauhan2889) [inline app/templates/components/tables/utilities/search-box.hbs:2]: we are using this component at some other places also. At mobile view we don't want changes. exmaple:
new speaker

You can use this condition _(and this.device.isMobile (eq this.session.currentRoute 'public.sessions'))_

**Review by sachinchauhan2889 — CHANGES_REQUESTED** [review]
there are slight more changes needed. 
1) there must be an animation or transition when we focus in and focus out search box. You can compare search bar on top of front page.
2) On focus in, search bar expand from left side and sort box & timezone box disappear and search box takes its area.

You can refer this https://github.com/fossasia/open-event-frontend/pull/6007

**Comment by iamareebjamal** [comment]
No animation required

**Comment by ranjitkshah** [comment]
Please review logic now @iamareebjamal

**Comment by maze-runnar** [comment]
by time and drop dropdown going in 2nd line.

File: app/components/widgets/time-zone-picker.hbs [hunk 3]
```diff
@@ -8,7 +8,7 @@
     @type="hidden"
     @value={{@timezone}} />
   <i class="dropdown icon"></i>
-  <div class="default text">{{@timezone}}</div>
+  <div class="default text time-zone-picker-text">{{@timezone}}</div>
```
Comment (ranjitkshah) [inline app/components/widgets/time-zone-picker.hbs:11]: not able to add this.device.isMobile condition here. any help?

**Comment by ranjitkshah** [comment]
updated screenshot:
image

**Comment by ranjitkshah** [comment]
@maze-runnar pls review

**Comment by ranjitkshah** [comment]
@iamareebjamal please re-review

**Comment by mariobehling** [comment]
Expected is that the search bar covers the complete width. The search magnifying lense icon should stay at the same place.

**Comment by ranjitkshah** [comment]
https://user-images.githubusercontent.com/61561415/104828365-ad3d2d00-588e-11eb-86a1-8695db24ab5c.mov

@mariobehling pls check now

**Comment by mariobehling** [comment]
Thank you. The height of the search bar seems not to be the same height as the timezone box. Please check.

Screenshot from 2021-01-17 21-30-50

**Comment by ranjitkshah** [comment]
update : 
now both have same height 36px
image
image

**Comment by ranjitkshah** [comment]
> update :
> now both have same height 36px
> image
> image

@mariobehling

**Review by mariobehling — DISMISSED** [review]
Nice. Works on my tests. Thank you.

File: app/components/tables/utilities/search-box.js [hunk 4]
```diff
@@ -17,4 +17,9 @@ export default class SearchBox extends Component {
     debounce(this, this.setSearchQueryLazily, value, this.debouncePeriod);
   }
 
+  @action
+  showSessionFields() {
+    this.sendAction('showSession');
```
Comment (iamareebjamal) [inline app/components/tables/utilities/search-box.js:22]: Why?

**Comment by iamareebjamal** [comment]
I don't understand the changes in this PR. Explain in the Tuesday meeting

**Comment by ranjitkshah** [comment]
> I don't understand the changes in this PR. Explain in the Tuesday meeting

now, logic fixed  @iamareebjamal   And also I don't know why I was doing this ; )

File: app/components/widgets/time-zone-picker.hbs [hunk 5]
```diff
@@ -8,7 +8,7 @@
     @type="hidden"
     @value={{@timezone}} />
   <i class="dropdown icon"></i>
-  <div class="default text">{{@timezone}}</div>
+  <div class="default text {{ if @device.isMobile "time-zone-picker-text" "" }}">{{@timezone}}</div>
```
Comment (maze-runnar) [inline app/components/widgets/time-zone-picker.hbs:11]:
```suggestion
  <div class="default text {{if @device.isMobile "time-zone-picker-text" "" }}">{{@timezone}}</div>
```

File: app/styles/libs/_helpers.scss [hunk 6]
```diff
@@ -127,6 +127,30 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
...
+
+.full-width {
+  width: 100%;
+}
+
+.time-zone-picker-text {
+  text-overflow: ellipsis;
+  overflow: hidden;
+  white-space: nowrap;
+  max-width: 88px;
```
Comment (maze-runnar) [inline app/styles/libs/_helpers.scss:151]: fix the sorting lint of scss. 
```suggestion
  max-width: 88px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
```

File: app/templates/components/tables/utilities/search-box.hbs [hunk 7]
```diff
@@ -1,4 +1,4 @@
 <div class="ui {{if (eq @size "large") '' 'small'}} icon input search-box">
-  <Input @type="text" @value={{readonly this.searchQuery}} @keyUp={{action 'setSearchQuery' value="target.value"}} placeholder="{{t "Search"}} ..." />
+  <Input  class="{{ if (and this.device.isMobile (eq this.session.currentRouteName 'public.sessions')) "expand-bar" "" }}" @type="text" @value={{readonly this.searchQuery}} @keyUp={{action 'setSearchQuery' value="target.value"}} onfocus={{ if (and this.device.isMobile (eq this.session.currentRouteName 'public.sessions')) (action @showSession)}} @focus-out={{if (and this.device.isMobile (eq this.session.currentRouteName 'public.sessions')) (action @showSession)}} placeholder="{{t "Search"}} ..." />
```
Comment (maze-runnar) [inline app/templates/components/tables/utilities/search-box.hbs:2]:
```suggestion
  <Input  class="{{if (and this.device.isMobile (eq this.session.currentRouteName 'public.sessions')) "expand-bar" "" }}" @type="text" @value={{readonly this.searchQuery}} @keyUp={{action 'setSearchQuery' value="target.value"}} onfocus={{if (and this.device.isMobile (eq this.session.currentRouteName 'public.sessions')) (action @showSession)}} @focus-out={{if (and this.device.isMobile (eq this.session.currentRouteName 'public.sessions')) (action @showSession)}} placeholder="{{t "Search"}} ..." />
```

**Comment by MeghalBisht** [comment]
A lot of changes were made for the schedule page after this PR was raised. Hence a lot of conflicts and redundant code is present. Hence closing this PR.
Also what I remember from the discussion with Areeb, he advised that too many changes were needed to achieve the result.
@ranjitkshah Please follow up in a new PR if needed.

**Pull request closed without merging**